### PR TITLE
UI testing issue21

### PR DESCRIPTION
**Pull request by MichaelPesce**
Added component tests using react-testing-library
Added e2e tests using Cypress
Added Github action to run component tests on push/PR
Cleaned up some frontend code

### CODE REVIEW

File: electron/ui/src/tests/FlowsheetsListTable.test.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { render, screen } from '@testing-library/react';
+import FlowsheetsListTable from "../components/FlowsheetsListTable/FlowsheetsListTable"
+import test_data from '../../../../backend/data/flowsheets/fake/data.json'
+import * as React from 'react'
+
+test('test flowsheets list table', () => {
```
Comment (dangunter) [inline electron/ui/src/tests/FlowsheetsListTable.test.js:6]: The METAB flowsheet get moved/renamed at some point. Not sure how to deal with that automatically

File: electron/ui/src/views/FlowsheetConfig/FlowsheetConfig.js [hunk 2]
```diff
@@ -134,6 +141,9 @@ export default function FlowsheetConfig() {
       setOpenSuccessSaveSnackbar(false);
```
Comment (dangunter) [inline electron/ui/src/views/FlowsheetConfig/FlowsheetConfig.js:141]: I think this is sort of cruft left over from the example the load/save is copied from?

File: electron/ui/src/views/FlowsheetConfig/FlowsheetConfig.js [hunk 3]
```diff
@@ -134,6 +141,9 @@ export default function FlowsheetConfig() {
...
     };
 
+    const handleErrorSnackbarClose = () => {
```
Comment (dangunter) [inline electron/ui/src/views/FlowsheetConfig/FlowsheetConfig.js:144]: ditto.. can we get rid of the "Snackbar" part of these names (in a separate PR)?

**Review by dangunter — APPROVED** [review]
Looks good. I would remove the package-lock.json file from the commit though, and consider a more complete .gitignore

File: electron/ui/src/views/FlowsheetConfig/FlowsheetConfig.js (hunk 2 above)
Comment (MichaelPesce) [reply electron/ui/src/views/FlowsheetConfig/FlowsheetConfig.js:141]: This is a little component that pops up briefly when a user saves configuration changes. I think it makes sense to keep this (or something like it) as a quick confirmation that changes were saved.

File: electron/ui/src/views/FlowsheetConfig/FlowsheetConfig.js (hunk 3 above)
Comment (MichaelPesce) [reply electron/ui/src/views/FlowsheetConfig/FlowsheetConfig.js:144]: I added this one as a pop up that opens when an error occurs when running the model (ie bad data input). Snackbar is just the name of the MUI component that pops up, but I can create a better name for it, as well as improve the error handling a bit

**Comment by dangunter** [comment]
It can we take “snack bar” out of the name? Seems weird.

**Pull request merged**